mongomock/aggregate.py

```diff
@@ -984,6 +984,28 @@ def _find_matches_for_depth(query):
                 found_items.add(new_match['_id'])
         return new_matches
 
+    def _recursive_get(match, nested_fields):
+        first_field = nested_fields[0]
+        head = match.get(first_field)
+        if len(nested_fields) == 1:
+            # final/last field reached
+            yield head
+        else:
```

Comment:
I suggest you return early in the branch above instead of adding the whole code in an `else` branch:

```py
if len(nested_fields) == 1:
  yield head
  return

# More to go: get remaining from list/dict.